src/Data/Functor/Foldable.hs

```diff
@@ -339,6 +341,28 @@ instance Corecursive [a] where
     Cons x (Right b) -> x : apo f b
     Nil -> []
 
+data TreeF a b = NodeF a [b]
+  deriving (Eq,Ord,Show,Read,Typeable
+#if HAS_GENERIC
+          , Generic
+#endif
+#if HAS_GENERIC1
+          , Generic1
+#endif
+          )
+
+instance Functor (TreeF a) where
```

Comment:
We need Foldable, Traversable. ... Bitraversable too. Like with `ListF`

Reply:
In fact we'd probably like the `Eq1` etc. too, but those are hairy, and I can add them afterwards.

Reply:
Thanks for the quick feedback!

I got some instance definitions for Foldable, ..., Bitraversable compiling.

I can take a stab at the Eq1... instance definitions, probably later this week.